### ISSUE TITLE
Separation of magnitude/phase for B1 field

### ISSUE THREAD

**Issue by jcohenadad** [shimming-toolbox/shimming-toolbox-matlab#110]
Should we also separate magnitude/phase with the data loader, the same way this is done with B0 field maps?

i would say yes. Are there any cons?

**Comment by po09i** [comment]
I like the idea of having B0 and B1 data stored and used the same way. I'm not aware of any cons.

**Comment by gaspardcereza** [comment]
A little bit more context: 
- For B1 mapping, the scanners does most of the work. I just need to reorganize the Dicoms (gather the corresponding phase and magnitude for each slice and each coil)  and scale them to get the B1 maps that will be used to compute the shim coefficient. These B1 maps are 4-dimensional (x, y, z, Coil). 
- For B0 mapping, we have magnitude and phase maps with different times and different echos. We thus have 5-dimensional data (x, y, z, echo, time). 

We decided to use the Nifti format for both B1 and B0 data before computing the corresponding maps. During the conversion from Dicom to Nifti, the B0 data are separated into 2 different folders (phase and magnitude). I don't specifically need that separation for my B1 data and dicom2bids doesn't separate them by default. 
I was thus wondering if it was really necessary to add a separating step before I compute my B1 maps, considering that I can can compute my maps without it. I personally prefer to have all the data stored in a single Nifti (I think it reduces the risks of losing data and makes them easier to handle) but I admit that it is also better to have some consistency in our data.

I'd be glad to have some feedback about it.

**Comment by evaalonsoortiz** [comment]
It's not clear to me if you are asking whether we should have magnitude and phase nifti files in separate folders (IMO: not necessary, we can have both in the same folder if you prefer that) or if it is ok to have magnitude and phase data as part of the same nifti file (although possible, I would say definitely no).

**Comment by jcohenadad** [comment]
@gaspardcereza i suggest you update the [flowchart](https://app.diagrams.net/#G1BqYuqOvTUyPTHhb3z19rktEv6lZU3ehM) to indicate the type of data you are acquiring under the triangle "B1 mapping". That way, the whole team is on the same page. 

> For B0 mapping, we have magnitude and phase maps with different times and different echos. We thus have 5-dimensional data (x, y, z, echo, time).

Are you talking about NIfTI data or matlab matrices? If the former: can you point us to the code that creates these 5-d NIfTI files?  If the latter: this is a non-problem (because this issues has to do with data loader, i.e., loading of DCM/NIfTI files).

> We decided to use the Nifti format for both B1 and B0 data before computing the corresponding maps. During the conversion from Dicom to Nifti, the B0 data are separated into 2 different folders (phase and magnitude). I don't specifically need that separation for my B1 data and dicom2bids doesn't separate them by default.
I was thus wondering if it was really necessary to add a separating step before I compute my B1 maps, considering that I can can compute my maps without it. I personally prefer to have all the data stored in a single Nifti (I think it reduces the risks of losing data and makes them easier to handle) but I admit that it is also better to have some consistency in our data.

I am a bit confused by your explanation.  There are 2 things to consider:
- dcm2nifti: does separate by folder (if we ask to do so)
- dcm2bids: does not separate by folder: will put the mag nifti file and the phase nifti file into the same fmap/ folder.

So, there is never a question of merging mag and phase into the same nifti file, and it should not be done.

Could you please elaborate/rephrase what you mean?

**Comment by gaspardcereza** [comment]
> It's not clear to me if you are asking whether we should have magnitude and phase nifti files in separate folders (IMO: not necessary, we can have both in the same folder if you prefer that) or if it is ok to have magnitude and phase data as part of the same nifti file (although possible, I would say definitely no).

I was asking the second option. Why is it preferable to have phase and magnitude in different Niftis ?

**Comment by jcohenadad** [comment]
> I was asking the second option. Why is it preferable to have phase and magnitude in different Niftis ?

because
- this is more common
- the image quantization could be different between phase and magnitude data
- most fieldmap software do expect 3D inputs (phase and/or mag)

**Comment by gaspardcereza** [comment]
> So, there is never a question of merging mag and phase into the same nifti file, and it should not be done.

dcm2niix puts them into the same nifti by default. I'll look for a way around it. Thanks for the explanation. It is indeed easier when written down 😉

**Comment by jcohenadad** [comment]
> dcm2niix puts them into the same nifti by default.

ah?

> I'll look for a way around it.

nope, that's not the right procedure 😉 . Can you please:
- post a new issue in this repository to declare this problem
- provide syntax + data for others to be able to reproduce the issue
- indicate your version of dcm2niix + OS

once issue is open you can look for a way around it and document your investigations.

**Comment by evaalonsoortiz** [comment]
Could you could share some dicom B1+ data with me? I'd like to test this out.

**Comment by gaspardcereza** [comment]
> Could you could share some dicom B1+ data with me? I'd like to test this out.

I used the standard rf maps from the MNI measurements (on duke/mri/mni-7t):
ACDC_SPINE7T_001_20_02_20-16_27_40-STD1_3_12_2_1107_5_2_0_79017\DEVELOPMENT_DR_COHENADAD_20200221_110129_199000\STANDARD_0106

**Comment by jcohenadad** [comment]
@gaspardcereza no please, do it the proper way, as i described here: https://github.com/shimming-toolbox/shimming-toolbox/issues/110#issuecomment-644785868

we all need to get the habit of properly documenting each issue, separately, and associate datasets to it (if <10MB, upload as zip). The present issue thread is specific to "Separation of magnitude/phase for B1 field", this has nothing to do with the dcm2niix error, which again, should be located in a different issue. Sorry to be annoying, but again, a good organization habit is key for an efficient development environment

**Comment by gaspardcereza** [comment]
> @gaspardcereza no please, do it the proper way, as i described here: [#110 (comment)](https://github.com/shimming-toolbox/shimming-toolbox/issues/110#issuecomment-644785868)

Yes I'm on it ! I'm just trying to gather info to make it clearer. I just wanted to give Eva the data if she wanted to have a quick look at what it does
